tests/openat.c

```diff
@@ -161,6 +167,158 @@ main(void)
 			perror_msg_and_fail("unlink");
 	}
 
+# ifdef TEST_SECONTEXT
+
+	char buf[PATH_MAX];
+	ssize_t n = readlink(self_fd0, buf, sizeof(buf) - 1);
+	if ((n > 0) && ((size_t) n < sizeof(buf) - 1)) {
+		buf[n] = '\0';
+		char *self_fd0_secontext = SECONTEXT_FILE(self_fd0);
+		char *target_secontext = SECONTEXT_FILE(buf);
+
+		fd = syscall(__NR_openat, cwd_fd, self_fd0, O_RDONLY);
+		printf("%s%s(%d%s, \"%s\"%s, O_RDONLY) = %s%s\n",
+		       my_secontext, "openat",
+		       cwd_fd, cwd_secontext,
+		       self_fd0, self_fd0_secontext,
+		       sprintrc(fd), target_secontext);
+		if (fd != -1)
+			close(fd);
+	}
+
+	const char *proc_self_paths[] = {
+		"/proc/self", "/proc/self/root/proc/self"
+	};
+
+	for (unsigned int i = 0; i < sizeof(proc_self_paths)/sizeof(*proc_self_paths); i++) {
+		/*
+		 * /proc/self substitution is special: /proc/self is a
+		 * symlink, which has context "proc_t", but strace
+		 * substitutes /proc/self by /proc/<pid>, which is a
+		 * directory, hence has context of the process itself.
+		 *
+		 * Additionally the SECONTEXT_PID_MY() appends a space,
+		 * instead of prepending it, so fix this here
+		 */
+		char *secontext = SECONTEXT_PID_MY();
+		if (strlen(secontext) > 0) {
+			char *c;
+			for (c = secontext + strlen(secontext) - 1; c > secontext; c--)
+				*c = *(c - 1);
+			*c = ' ';
+		}
+
+		fd = syscall(__NR_openat, cwd_fd, proc_self_paths[i], O_RDONLY);
+		printf("%s%s(%d%s, \"%s\"%s, O_RDONLY) = %s%s\n",
+		       my_secontext, "openat",
+		       cwd_fd, cwd_secontext,
+		       proc_self_paths[i], secontext,
+		       sprintrc(fd), secontext);
```

Comment:
For some reason this doesn't pass CI on ubuntu-22.04 runners with the following diagnostics:
```diff
-[unconfined] openat(4, "/proc/self" [unconfined], O_RDONLY) = 5 [unconfined]
-[unconfined] openat(4, "/proc/self/root/proc/self" [unconfined], O_RDONLY) = 5 [unconfined]
+[unconfined] openat(4, "/proc/self", O_RDONLY) = 5
+[unconfined] openat(4, "/proc/self/root/proc/self", O_RDONLY) = 5
```